### ISSUE TITLE
 Failed to execute 'getImageData' on 'CanvasRenderingContext2D'

### ISSUE THREAD

**Issue by Awatatah** [Mawi137/ngx-image-cropper#134]
For some reason I am getting this error on `[imageBase64]="dataUrl"`. It doesn't seem to be consistent, example, if I click  a Modal that contains the `<image-cropper></image-cropper>` element, I have a function that changes the `dataUrl` to a new value. It periodically causes the error when I open the Modal and `dataUrl` is changed.

> ERROR DOMException: Failed to execute 'getImageData' on 'CanvasRenderingContext2D': The source width is 0.
    at resizeCanvas (http://localhost:4200/vendor.js:72421:23)
    at ImageCropperComponent.push../node_modules/ngx-image-cropper/fesm5/ngx-image-cropper.js.ImageCropperComponent.crop (http://localhost:4200/vendor.js:73140:21)
    at ImageCropperComponent.push../node_modules/ngx-image-cropper/fesm5/ngx-image-cropper.js.ImageCropperComponent.doAutoCrop (http://localhost:4200/vendor.js:73104:18)
    at ImageCropperComponent.push../node_modules/ngx-image-cropper/fesm5/ngx-image-cropper.js.ImageCropperComponent.resetCropperPosition (http://localhost:4200/vendor.js:72849:14)
    at ImageCropperComponent.push../node_modules/ngx-image-cropper/fesm5/ngx-image-cropper.js.ImageCropperComponent.ngOnChanges (http://localhost:4200/vendor.js:72605:18)
    at checkAndUpdateDirectiveInline (http://localhost:4200/vendor.js:42344:19)
    at checkAndUpdateNodeInline (http://localhost:4200/vendor.js:43612:20)
    at checkAndUpdateNode (http://localhost:4200/vendor.js:43574:16)
    at debugCheckAndUpdateNode (http://localhost:4200/vendor.js:44207:38)
    at debugCheckDirectivesFn (http://localhost:4200/vendor.js:44167:13)

I could probably resolve by adding a `(click)` function to disable the `ImageCropperModule` when the Modal is closed but I do not know how to do this. 

Basically I would like to destroy the `ImageCropperModule` instance and reinitialize it on command. Is there a way to do this?

**Comment by Awatatah** [comment]
I resolved by adding a `(click)="fileChangeEvent(null)"` to the Modal close btn and the follwing

In Component
```
fileChangeEvent(event: any): void {
    	this.imageChangedEvent = event;
}
```

In html
```
<image-cropper #cropper
    ...
    [imageChangedEvent]="imageChangedEvent"
    ...
></image-cropper>
```

**Comment by albert-hoffman** [comment]
@Awatatah : [imageChangedEvent]="imageChangedEvent" here "imageChangedEvent" is not defined, how did you do that then?
